frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -540,7 +541,7 @@ function ProblemDisplay(props: ProblemDisplayParams) {
         <SmallHeaderText>Options</SmallHeaderText>
         <SettingsContainer>
           <ApprovalContainer
-            show={editMode}
+            show={editMode && account?.role === AccountRole.Admin}
           >
             <ToggleButton
               onChangeFunction={() => handleApprovalChange(!newProblem.approval)}
```

Comment:
Yeah I think that's a great idea - would be the ideal situation to have a marketplace like that in the future 

About renaming - given that these might actually be treated as verified problems rather than public, what are your thoughts on naming the variable `public` as opposed to something like `verified`?

Reply:
I think that works - we'll need Public and Private eventually and both have essentially the same functionality right now, so I'll say you can choose which one you want to switch this to - `public` and `verified` both work.

Reply:
In my mind, I did see `verified` as something we might just change in the database to be honest, and then change the `public` toggle such that _anybody_ can access it, but both solutions work by me.